lib/Doctrine/ORM/Id/UuidGenerator.php

```diff
@@ -22,10 +26,16 @@ public function __construct()
             '%s is deprecated with no replacement, use an application-side generator instead',
             self::class
         );
+
+        if (! method_exists(AbstractPlatform::class, 'getGuidExpression')) {
+            throw NotSupported::createForDbal3();
+        }
     }
 
     /**
      * {@inheritDoc}
+     *
+     * @throws NotSupported
      */
     public function generate(EntityManager $em, $entity)
     {
```

Comment:
Apparently this method is no longer covered: https://app.codecov.io/gh/doctrine/orm/compare/8898/changes
Why is that?

Reply:
If the coverage is taken from tests with doctrine/dbal:3.0, this function will not be tested as the class itself will be never initialised.

Reply:
@greg0ire - it looks like the possible reason might be the same name `--coverage-clover=coverage-cache.xml` used for the tests with dbal2 and dbal3.

Reply:
It's close to that, but that's not it, I think you can have the same name as long as it differs when it's uploaded (so the upload step needs to be modified)

Reply:
Or maybe I'm mistaken, looks like I already took care of setting different names when introducing the DBAL 3 build

Reply:
> If the coverage is taken from tests with doctrine/dbal:3.0, this function will not be tested as the class itself will be never initialised.

The coverage is supposed to be merged from multiple reports

Reply:
The list can be seen here: https://github.com/doctrine/orm/pull/8898/checks?check_run_id=3343305825#step:3:21

I think some might be missing indeed.

Reply:
Ah but the DBAL3 tests don't even run right now, so there will not be a report from them. The DBAL2 tests are running though, and are supposed to cover that path.

Reply:
The job supposed to cover that path are the mysql ones, and they are present… not sure what to think :thinking:

Reply:
Independent from this PR: Shouldn't this also be covered by the other platforms?

Reply:
:thinking: good question, might be worth removing that call to `self::markTestSkipped` and see on which other platforms it passes. The situation might have evolved since this test was written in 2012.